### PR TITLE
Remove zalando problems: Upgrade to Spring Boot 3

### PR DESCRIPTION
**Pull request by DamnClin**
Close https://github.com/jhipster/jhipster-lite/issues/4111
Close https://github.com/jhipster/jhipster-lite/issues/4066
Close https://github.com/jhipster/jhipster-lite/issues/4351
Close https://github.com/jhipster/jhipster-lite/issues/3975
Close https://github.com/jhipster/jhipster-lite/issues/3750

### CODE REVIEW

**Comment by DamnClin** [comment]
Spring cloud release for Spring Boot 3 is not there yet: https://github.com/spring-cloud/spring-cloud-release/wiki/Spring-Cloud-2022.0-Release-Notes, that leaves us with the following options: 

* Wait a Spring Cloud release;
* Use Spring Cloud Release Candidate (I don't think this is acceptable...);
* Disable Spring cloud related modules (`spring-cloud`, `consul`, `eureka-client`, `gateway`) and wait for release.

File: src/main/resources/generator/server/springboot/mvc/security/jwt/authentication/main/infrastructure/primary/SecurityConfiguration.java.mustache [hunk 1]
```diff
@@ -81,19 +75,20 @@ class SecurityConfiguration {
...
-        .antMatchers("/management/prometheus").permitAll()
-        .antMatchers("/management/**").hasAuthority(Role.ADMIN.key())
+      .authorizeHttpRequests()
+        .requestMatchers("/api/authenticate").permitAll()
+        .requestMatchers("/api/register").permitAll()
+        .requestMatchers("/api/activate").permitAll()
+        .requestMatchers("/api/account/reset-password/init").permitAll()
+        .requestMatchers("/api/account/reset-password/finish").permitAll()
+        .requestMatchers("/api/admin/**").hasAuthority(Role.ADMIN.key())
+        .requestMatchers("/api/**").authenticated()
```
Comment (DamnClin) [inline src/main/resources/generator/server/springboot/mvc/security/jwt/authentication/main/infrastructure/primary/SecurityConfiguration.java.mustache:85]: Got lots of doubt on that...

File: src/main/resources/generator/server/springboot/mvc/security/jwt/authentication/main/infrastructure/primary/SecurityConfiguration.java.mustache [hunk 2]
```diff
@@ -81,19 +75,20 @@ class SecurityConfiguration {
...
+        .requestMatchers("/management/health").permitAll()
+        .requestMatchers("/management/health/**").permitAll()
+        .requestMatchers("/management/info").permitAll()
+        .requestMatchers("/management/prometheus").permitAll()
+        .requestMatchers("/management/**").hasAuthority(Role.ADMIN.key())
+        .requestMatchers("/**").authenticated()
```
Comment (DamnClin) [inline src/main/resources/generator/server/springboot/mvc/security/jwt/authentication/main/infrastructure/primary/SecurityConfiguration.java.mustache:91]: Got lots of doubt on that...

File: src/main/resources/generator/server/springboot/mvc/security/oauth2/core/main/infrastructure/primary/SecurityConfiguration.java.mustache [hunk 3]
```diff
@@ -92,16 +85,17 @@ public class SecurityConfiguration {
...
+      .requestMatchers("/api/authenticate").permitAll()
+      .requestMatchers("/api/auth-info").permitAll()
+      .requestMatchers("/api/admin/**").hasAuthority(Role.ADMIN.key())
+      .requestMatchers("/api/**").authenticated()
+      .requestMatchers("/management/health").permitAll()
+      .requestMatchers("/management/health/**").permitAll()
+      .requestMatchers("/management/info").permitAll()
+      .requestMatchers("/management/prometheus").permitAll()
+      .requestMatchers("/management/**").hasAuthority(Role.ADMIN.key())
+      .requestMatchers("/**").authenticated()
```
Comment (DamnClin) [inline src/main/resources/generator/server/springboot/mvc/security/oauth2/core/main/infrastructure/primary/SecurityConfiguration.java.mustache:98]: Got lots of doubt on that...

File: src/main/resources/generator/server/springboot/mvc/security/jwt/authentication/main/infrastructure/primary/SecurityConfiguration.java.mustache [hunk 4]
```diff
@@ -81,19 +75,20 @@ class SecurityConfiguration {
...
+        .requestMatchers("/api/account/reset-password/init").permitAll()
+        .requestMatchers("/api/account/reset-password/finish").permitAll()
+        .requestMatchers("/api/admin/**").hasAuthority(Role.ADMIN.key())
+        .requestMatchers("/api/**").authenticated()
+        .requestMatchers("/management/health").permitAll()
+        .requestMatchers("/management/health/**").permitAll()
+        .requestMatchers("/management/info").permitAll()
+        .requestMatchers("/management/prometheus").permitAll()
+        .requestMatchers("/management/**").hasAuthority(Role.ADMIN.key())
+        .anyRequest().authenticated()
```
Comment (DamnClin) [inline src/main/resources/generator/server/springboot/mvc/security/jwt/authentication/main/infrastructure/primary/SecurityConfiguration.java.mustache:91]: Got lots of doubts on that. Should we delete the /api/** one?

File: src/main/resources/generator/server/springboot/mvc/security/oauth2/core/main/infrastructure/primary/SecurityConfiguration.java.mustache [hunk 5]
```diff
@@ -92,16 +85,17 @@ public class SecurityConfiguration {
...
+      .requestMatchers("/api/authenticate").permitAll()
+      .requestMatchers("/api/auth-info").permitAll()
+      .requestMatchers("/api/admin/**").hasAuthority(Role.ADMIN.key())
+      .requestMatchers("/api/**").authenticated()
+      .requestMatchers("/management/health").permitAll()
+      .requestMatchers("/management/health/**").permitAll()
+      .requestMatchers("/management/info").permitAll()
+      .requestMatchers("/management/prometheus").permitAll()
+      .requestMatchers("/management/**").hasAuthority(Role.ADMIN.key())
+      .anyRequest().authenticated()
```
Comment (DamnClin) [inline src/main/resources/generator/server/springboot/mvc/security/oauth2/core/main/infrastructure/primary/SecurityConfiguration.java.mustache:98]: Got lots of doubts on that. Should we delete the /api/** one?

File: src/main/resources/generator/server/springboot/mvc/security/jwt/authentication/main/infrastructure/primary/SecurityConfiguration.java.mustache (hunk 4 above)
Comment (pascalgrimaud) [reply src/main/resources/generator/server/springboot/mvc/security/jwt/authentication/main/infrastructure/primary/SecurityConfiguration.java.mustache:92]: with this, the front part can be displayed unless you are authenticated, so we need to work on this

**Comment by DamnClin** [comment]
Claimed https://opencollective.com/generator-jhipster/expenses/113202

**Comment by pascalgrimaud** [comment]
@DamnClin : approved

**Pull request merged**